samples/excel/30-events/events-comment-event-handler.yaml

```diff
@@ -0,0 +1,195 @@
+            await Excel.run(async (context) => {
+                const selectedRange = context.workbook.getSelectedRange();
+
+                // Note that an InvalidArgument error will be thrown if multiple cells are selected.
+                context.workbook.comments.add(selectedRange, "TODO: add headers here.");
+
+                await context.sync();
+            });
+        }
+
+        async function addCommentToCell() {
+            await Excel.run(async (context) => {
+                const sheet = context.workbook.worksheets.getItem("Comments");
+
+                // Note that an InvalidArgument error will be thrown if multiple cells are passed to `comment.add`.
+                sheet.comments.add("A2", "TODO: add data.");
+                await context.sync();
+            });
+        }
+
+        async function editComment() {
+            await Excel.run(async (context) => {
+                const sheet = context.workbook.worksheets.getItem("Comments");
+                const comment = sheet.comments.getItemAt(0);
+                comment.content = "PLEASE add headers here.";
+                await context.sync();
+            });
+        }
+
+        async function deleteComment() {
+            await Excel.run(async (context) => {
+                context.workbook.comments.getItemByCell("Comments!A2").delete();
+                await context.sync();
+            });
+        }
+
+        async function setup() {
+            await Excel.run(async (context) => {
+                context.workbook.worksheets.getItemOrNullObject("Comments").delete();
+                const sheet = context.workbook.worksheets.add("Comments");
+
+                sheet.activate();
+                await context.sync();
+            });
+        }
+
+        /** Default helper for invoking an action and handling errors. */ 
+        async function tryCatch(callback) {
+            try {
+                await callback();
+            } catch (error) {
+                // Note: In a production add-in, you'd want to notify the user through your add-in's UI.
+                console.error(error);
+            }
+        }
+    language: typescript
+template:
+    content: |-
+        <section class="ms-font-m">
+            <p>This sample shows how to add, edit, and remove comments.</p>
+        </section>
+        <section class="setup ms-font-m">
+            <h3>Setup</h3>
+            <button id="setup" class="ms-Button">
+                <span class="ms-Button-label">Create a worksheet</span>
+            </button>
+            <p></p>
+            <button id="registerEventHandlers" class="ms-Button">
+                <span class="ms-Button-label">Register event handlers</span>
+            </button>
+        </section>
+        <section class="samples ms-font-m">
+            <h3>Try it out</h3>
+            <p>To better view the comment changes, open the Comments pane by selecting <b>Show Comments</b> from the
```

Comment:
I'm not sure this note is as interesting in this sample, versus the original comment basics one. Your call on whether to keep it.

Reply:
My main concern about this sentence is that the Comments pane conflicts with the Script Lab task pane -- a user needs to click between two overlapping panes repeatedly to see the comment changes (in either sample). I left the sentence in here because it might be helpful for a developer less familiar with the little purple comments flag that appears on cells with comments.

Reply:
I adjusted this sentence to better clarify its purpose in this sample (to make sure the user is aware of the Comments pane).